eng/common/docgeneration/Generate-DocIndex.ps1

```diff
@@ -0,0 +1,195 @@
+            if ($displayName) {
+                Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact`n##### ($displayName)"
+            }
+            else {
+                Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact"
+            }
+        }
+        else {
+            Add-Content -Path "$($YmlPath)/toc.yml" -Value "- name: ${serviceName}`r`n  href: ${fileName}.md"
+            New-Item -Path $YmlPath -Name "${fileName}.md" -Force
+            if ($displayName) {
+                Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact`n##### ($displayName)"
+            }
+            else {
+                Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact"
+            }
+            $visitedService[$serviceName] = $true
+        }
+    }
+
+    # Generate toc homepage.
+    LogDebug "Creating Site Title and Navigation..."
+    New-Item -Path "${DocOutDir}" -Name "toc.yml" -Force
+    Add-Content -Path "${DocOutDir}/toc.yml" -Value "- name: Azure SDK for $lang APIs`r`n  href: api/`r`n  homepage: api/index.md"
+
+    LogDebug "Copying root markdowns"
+    Copy-Item "$($RepoRoot)/README.md" -Destination "${DocOutDir}/api/index.md" -Force
+    Copy-Item "$($RepoRoot)/CONTRIBUTING.md" -Destination "${DocOutDir}/api/CONTRIBUTING.md" -Force
+
+    LogDebug "Building site..."
+    #& $($DocFx) build "${DocOutDir}/docfx.json"
+    # The line below is used for testing in local
+    docfx build "${DocOutDir}/docfx.json"
+    Copy-Item "${DocGenDir}/assets/logo.svg" -Destination "${DocOutDir}/_site/" -Force    
+}
+
+function Mutate-Files {
+    Param (
+        [Parameter(Mandatory=$true)] [String]$lang,
+        [Parameter(Mandatory=$true)] [String]$indexhtmlloc,
+        [Parameter(Mandatory=$false)] [String]$packageRegex,
+        [Parameter(Mandatory=$false)] [String]$regexReplacement
+    )
+    # Update docfx.json
+    $docfxContent = Get-Content -Path $DocfxJsonPath -Raw
+    $docfxContent = $docfxContent -replace "`"_appTitle`": `"`"", "`"_appTitle`": `"$appTitle`""
+    $docfxContent = $docfxContent -replace "`"_appFooter`": `"`"", "`"_appFooter`": `"$appTitle`""
+    Set-Content -Path $DocfxJsonPath -Value $docfxContent
+    # Update main.js var lang
+    $mainJsContent = Get-Content -Path $MainJsPath -Raw
+    $mainJsContent = $mainJsContent -replace "var SELECTED_LANGUAGE = ''", "var SELECTED_LANGUAGE = '$lang'"
+    # Update main.js var index html
+    $mainJsContent = $mainJsContent -replace "var INDEX_HTML = ''", "var INDEX_HTML = '$indexhtmlloc'"
+    # Update main.js package regex and replacement
+    $mainJsContent = $mainJsContent -replace "var PACKAGE_REGEX = ''", "var PACKAGE_REGEX = '$packageRegex'"
+    $mainJsContent = $mainJsContent -replace "var PACKAGE_REPLACEMENT = ''", "var PACKAGE_REPLACEMENT = '$regexReplacement'"
+    Set-Content -Path $MainJsPath -Value $mainJsContent
+}
+
+if (Test-Path "function:$GetGithubIoDocIndexFn")
```

Comment:
```suggestion
if ($GetGithubIoDocIndexFn -and Test-Path "function:$GetGithubIoDocIndexFn")
```

@chidozieononiwu and I figured out recently that we need to also check for the existence of this variable before testing the path, otherwise the test path will always return true when the variable isn't set to anything.